src/Raven.Server/Documents/Indexes/IndexStore.cs

```diff
@@ -628,6 +634,9 @@ private Index HandleStaticIndexChange(string name, IndexDefinition definition)
                 if (creationOptions == IndexCreationOptions.Update)
                 {
                     Debug.Assert(currentIndex != null);
+                    var definitionClone = new IndexDefinition();
+                    definition.CopyTo(definitionClone);
+                    definition = definitionClone;
```

Comment:
Can you please elaborate on this fix? I'd like to understand what was going on there.

Reply:
The bug happened because we modified the record index definitions on this method instead of changing only the definitions of of the local (node) indexes.

Reply:
The definition instance is from the database record we don't want to modify here the database record directly so we clone.

What happened is we modified the database record directly and change the `definition.Name` to be `ReplacementOf/...`, but we used the same database record elsewhere.. 
here: https://github.com/ravendb/ravendb/blob/5a1294432f77cb456b5a123b3c118b0e2fbf4ce4/src/Raven.Server/Documents/DocumentDatabase.cs#L321
and here: 
https://github.com/ravendb/ravendb/blob/5a1294432f77cb456b5a123b3c118b0e2fbf4ce4/src/Raven.Server/Documents/DocumentDatabase.cs#L355 

So the second time when we enter this method of `HandleStaticIndexChange` we already had the `ReplacementOf/...` in the name and added another `ReplacementOf/`